extensions/cornerstone/src/services/ToolService/ToolService.ts

```diff
@@ -0,0 +1,152 @@
+import { Types as OhifTypes, ServicesManager, PubSubService } from '@ohif/core';
+import { eventTarget } from '@cornerstonejs/core';
+import { Enums as cstEnums } from '@cornerstonejs/tools';
+import { CornerstoneServices } from '../../types';
+
+const EVENTS = {
+  TOOL_ACTIVATED: 'event::cornerstone::toolservice:toolActivated',
+  PRIMARY_TOOL_ACTIVATED:
+    'event::cornerstone::toolservice:primaryToolActivated',
+};
+
+class ToolService extends PubSubService {
+  static REGISTRATION = {
+    name: 'toolService',
+    altName: 'ToolService',
+    create: ({
+      servicesManager,
+    }: OhifTypes.Extensions.ExtensionParams): ToolService => {
+      return new ToolService({ servicesManager });
+    },
+  };
+
+  readonly servicesManager: ServicesManager;
+  readonly EVENTS = EVENTS;
+
+  constructor({ servicesManager }) {
+    super(EVENTS);
+
+    this.servicesManager = servicesManager;
+    this._initToolService();
+  }
+
+  private _initToolService() {
+    eventTarget.addEventListener(
+      cstEnums.Events.TOOL_ACTIVATED,
+      this._onToolActivated
+    );
+  }
+
+  public destroy = () => {
+    eventTarget.removeEventListener(
+      cstEnums.Events.TOOL_ACTIVATED,
+      this._onToolActivated
+    );
+  };
+
+  private _onToolActivated = evt => {
+    const { toolName, toolBindingsOptions } = evt.detail;
+    const isPrimaryTool = toolBindingsOptions.bindings?.some(
+      binding => binding.mouseButton === cstEnums.MouseBindings.Primary
+    );
+
+    this._broadcastEvent(EVENTS.TOOL_ACTIVATED, {
+      toolName,
+      toolBindingsOptions,
+    });
+
+    if (isPrimaryTool) {
+      this._broadcastEvent(EVENTS.PRIMARY_TOOL_ACTIVATED, { toolName });
+    }
+  };
+
+  public getActivePrimaryMouseButtonTool(toolGroupId?: string): string {
+    const { toolGroupService } = this.servicesManager
+      .services as CornerstoneServices;
+
+    const toolGroup = toolGroupService.getToolGroup(toolGroupId);
+
+    return toolGroup?.getActivePrimaryMouseButtonTool();
+  }
```

Comment:
this is something that I think toolGroupService should handle not toolService

Reply:
Moved to `toolGroupService`